hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NonFSOTaskHandler.java

```diff
@@ -0,0 +1,133 @@
+package org.apache.hadoop.ozone.recon.tasks;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.WithParentObjectId;
+import org.apache.hadoop.ozone.recon.spi.ReconNamespaceSummaryManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.*;
+
+public class NonFSOTaskHandler extends NSSummaryTask {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(NonFSOTaskHandler.class);
+
+  @Inject
+  public NonFSOTaskHandler(ReconNamespaceSummaryManager
+                            reconNamespaceSummaryManager) {
+    super(reconNamespaceSummaryManager);
+  }
+
+  public Collection<String> getTaskTables() {
+    return Arrays.asList(new String[]{KEY_TABLE});
+  }
+
+  @Override
+  public String getTaskName() {
+    return "NonFSOTaskHandler";
+  }
+
+  @Override
+  public Pair<String, Boolean> process(OMUpdateEventBatch events) {
+    Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
+    final Collection<String> taskTables = getTaskTables();
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, ? extends
+          WithParentObjectId> omdbUpdateEvent = eventIterator.next();
+      OMDBUpdateEvent.OMDBUpdateAction action = omdbUpdateEvent.getAction();
+
+      // we only process updates on OM's KeyTable
+      String table = omdbUpdateEvent.getTable();
+      boolean updateOnKeyTable = table.equals(KEY_TABLE);
+      if (!taskTables.contains(table)) {
+        continue;
+      }
+
+      String updatedKey = omdbUpdateEvent.getKey();
+
+      try {
+        if(updateOnKeyTable) {    //check whether we are having key or directory
```

Comment:
This test isn't necessary.  Its always going to be the key table right??

Reply:
Yeah, I was thinking about removing it entirely as well as the list with the tables.